### ISSUE TITLE
Cannot start CI testing for kfserving

### ISSUE THREAD

**Issue by jinchihe** [kubeflow/testing#607]
Cannot start CI testing for kfserving, while presubmit trigger, the error will be as beflow details:

```
+ /usr/local/bin/checkout.sh /src
+ SRC_DIR=/src
+ mkdir -p /src/kubeflow
+ git clone --depth=2 --no-single-branch https://github.com/kubeflow/kfserving.git /src/kubeflow/kfserving
Cloning into '/src/kubeflow/kfserving'...
Checking out files:  38% (7172/18800)   
Checking out files:  39% (7332/18800)   
Checking out files:  40% (7520/18800)   
Checking out files:  41% (7708/18800)   
Checking out files:  42% (7896/18800)   
Checking out files:  43% (8084/18800)   
Checking out files:  44% (8272/18800)   
Checking out files:  45% (8460/18800)   
Checking out files:  46% (8648/18800)   
Checking out files:  47% (8836/18800)   
Checking out files:  48% (9024/18800)   
Checking out files:  49% (9212/18800)   
Checking out files:  50% (9400/18800)   
Checking out files:  51% (9588/18800)   
Checking out files:  52% (9776/18800)   
Checking out files:  53% (9964/18800)   
Checking out files:  54% (10152/18800)   
Checking out files:  55% (10340/18800)   
Checking out files:  56% (10528/18800)   
Checking out files:  57% (10716/18800)   
Checking out files:  58% (10904/18800)   
Checking out files:  59% (11092/18800)   
Checking out files:  60% (11280/18800)   
Checking out files:  61% (11468/18800)   
Checking out files:  62% (11656/18800)   
Checking out files:  63% (11844/18800)   
Checking out files:  64% (12032/18800)   
Checking out files:  65% (12220/18800)   
Checking out files:  66% (12408/18800)   
Checking out files:  67% (12596/18800)   
Checking out files:  68% (12784/18800)   
Checking out files:  69% (12972/18800)   
Checking out files:  70% (13160/18800)   
Checking out files:  70% (13254/18800)   
Checking out files:  71% (13348/18800)   
Checking out files:  72% (13536/18800)   
Checking out files:  73% (13724/18800)   
Checking out files:  74% (13912/18800)   
Checking out files:  75% (14100/18800)   
Checking out files:  76% (14288/18800)   
Checking out files:  77% (14476/18800)   
Checking out files:  78% (14664/18800)   
Checking out files:  79% (14852/18800)   
Checking out files:  80% (15040/18800)   
Checking out files:  81% (15228/18800)   
Checking out files:  82% (15416/18800)   
Checking out files:  83% (15604/18800)   
Checking out files:  84% (15792/18800)   
Checking out files:  85% (15980/18800)   
Checking out files:  86% (16168/18800)   
Checking out files:  87% (16356/18800)   
Checking out files:  88% (16544/18800)   
Checking out files:  89% (16732/18800)   
Checking out files:  90% (16920/18800)   
Checking out files:  91% (17108/18800)   
Checking out files:  92% (17296/18800)   
Checking out files:  93% (17484/18800)   
Checking out files:  94% (17672/18800)   
Checking out files:  95% (17860/18800)   
Checking out files:  96% (18048/18800)   
Checking out files:  97% (18236/18800)   
Checking out files:  98% (18424/18800)   
Checking out files:  99% (18612/18800)   
Checking out files: 100% (18800/18800)   
Checking out files: 100% (18800/18800), done.
+ echo Job Name = kubeflow-kfserving-presubmit
+ REPO_DIR=/src/kubeflow/kfserving
+ cd /src/kubeflow/kfserving
+ '[' '!' -z 540 ']'
+ git fetch origin pull/540/head:pr
Job Name = kubeflow-kfserving-presubmit
From https://github.com/kubeflow/kfserving
 * [new ref]         refs/pull/540/head -> pr
+ '[' '!' -z bd418ce9d61db5462698198cde375833cac2442b ']'
+ git checkout bd418ce9d61db5462698198cde375833cac2442b
Note: checking out 'bd418ce9d61db5462698198cde375833cac2442b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at bd418ce... Default workers to 1
+ git submodule init
+ git submodule update
++ git describe --tags --always --dirty
/src/kubeflow/kfserving is at bd418ce
+ echo /src/kubeflow/kfserving is at bd418ce
+ git submodule
+ git status
HEAD detached at bd418ce
nothing to commit, working directory clean
+ IFS=';'
+ read -ra REPOS
REPOS=
+ echo REPOS=
+ '[' '!' -d /src/kubeflow/testing ']'
+ git clone --depth=1 https://github.com/kubeflow/testing.git /src/kubeflow/testing
Cloning into '/src/kubeflow/testing'...
+ python -m kubeflow.testing.run_e2e_workflow --project=kubeflow-ci --zone=us-east1-d --cluster=kubeflow-testing --bucket=kubernetes-jenkins --config_file=/src/kubeflow/kfserving/prow_config.yaml --repos_dir=/src
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 62, in <module>
    from kubernetes import client as k8s_client
  File "/usr/local/lib/python2.7/dist-packages/kubernetes/__init__.py", line 20, in <module>
    import kubernetes.config
  File "/usr/local/lib/python2.7/dist-packages/kubernetes/config/__init__.py", line 17, in <module>
    from .kube_config import (list_kube_config_contexts, load_kube_config,
  File "/usr/local/lib/python2.7/dist-packages/kubernetes/config/kube_config.py", line 23, in <module>
    import adal
  File "/usr/local/lib/python2.7/dist-packages/adal/__init__.py", line 34, in <module>
    from .authentication_context import AuthenticationContext
  File "/usr/local/lib/python2.7/dist-packages/adal/authentication_context.py", line 31, in <module>
    from .authority import Authority
  File "/usr/local/lib/python2.7/dist-packages/adal/authority.py", line 34, in <module>
    import requests
  File "/usr/local/lib/python2.7/dist-packages/requests/__init__.py", line 84, in <module>
    from urllib3.contrib import pyopenssl
  File "/usr/local/lib/python2.7/dist-packages/urllib3/contrib/pyopenssl.py", line 46, in <module>
    import OpenSSL.SSL
  File "/usr/lib/python2.7/dist-packages/OpenSSL/__init__.py", line 8, in <module>
    from OpenSSL import rand, crypto, SSL
  File "/usr/lib/python2.7/dist-packages/OpenSSL/SSL.py", line 118, in <module>
    SSL_ST_INIT = _lib.SSL_ST_INIT
AttributeError: 'module' object has no attribute 'SSL_ST_INIT'

```

**Comment by jinchihe** [comment]
@jlewi Could please someone can help to take a look? Thanks.

**Comment by jlewi** [comment]
It looks like the tag

```
gcr.io/kubeflow-ci/test-worker:latest 
```

Was recently updated. 
It's now pointing to:
https://pantheon.corp.google.com/gcr/images/kubeflow-ci/GLOBAL/test-worker@sha256:ca9d9d7b59ebbc516a214c02f4c9c312ee0055286692f2ad5a3b3189a8496565/details?tab=info

**Comment by jlewi** [comment]
There are a couple other images after that the there is an image from Dec 12.
gcr.io/kubeflow-ci/test-worker@sha256:fbfb9141c02e9a50e3679f84637fc2d7c142fe9eaa6b9f55a3c9f8a710000837 

I searched our issues. I think the most recent mention of an image update was
https://github.com/kubeflow/testing/issues/501#issuecomment-547863870

**Comment by jlewi** [comment]
Tried: gcr.io/kubeflow-ci/test-worker@sha256:fbfb9141c02e9a50e3679f84637fc2d7c142fe9eaa6b9f55a3c9f8a710000837

```
from OpenSSL import rand, crypto, SSL
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named OpenSSL
>>> import OpenSSL.SSL
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named OpenSSL.SSL
```

**Comment by jlewi** [comment]
Try:

```
gcr.io/kubeflow-ci/test-worker:v20191029-fd24db8-e3b0c4
```

Same error but the following works

```
from kubernetes import client as k8s_client
```

That also works on 
gcr.io/kubeflow-ci/test-worker@sha256:fbfb9141c02e9a50e3679f84637fc2d7c142fe9eaa6b9f55a3c9f8a710000837

It looks like the debug-worker-0 pod which was started 15 days ago is using 
gcr.io/kubeflow-ci/test-worker@sha256:fbfb9141c02e9a50e3679f84637fc2d7c142fe9eaa6b9f55a3c9f8a710000837

So lets roll back to that.

**Comment by jlewi** [comment]
@jinchihe You might want to think about updating your E2E tests to pin to a specific image. We could do this in a variety of places

1. The individual E2E test
1. In prow_config.yaml for every repository
1. In `run_e2e_workflow.py`
1. In a YAML file in kubeflow/testing

I think it makes the most sense to pin the image inside a config file in kubeflow/testing. This way at least when we update the image we leave a clear paper trail in git.

**Comment by jinchihe** [comment]
@jlewi  Thanks very much for the quick response!

**Comment by jlewi** [comment]
@gabrielwen were you building and retagging new worker images as part of the fix for kubeflow/kfctl#228?

**Comment by gabrielwen** [comment]
mm, I've seen this error, actually.  Yes I was rebuilding the image but I thought `make build-gcb` doesn't tag the image as latest?
